lib/ansible/module_utils/tacp_ansible/tacp_utils.py

```diff
@@ -148,13 +164,13 @@ def delete(self, uuid):
 class ApplicationResource(Resource):
     resource_class = tacp.ApplicationsApi
 
-    def filter(self, **filters):
-        return self.api.get_applications_using_get(
-            **self.get_filters_kws(**filters)
-        )
+    filter_method = "get_applications_using_get"
 
     def get_by_uuid(self, uuid):
-        return self.api.get_application_using_get(uuid)
+        resource = self.api.get_application_using_get(uuid)
```

Comment:
You should do the same with get_by_uuid now :-)
Have a uuid_method field which acts like filter_method.

Reply:
wow yeah this cleans it up a LOT - good call!